core/src/module/usModule.cpp

```diff
@@ -119,15 +121,37 @@ bool Module::IsLoaded() const
 
 void Module::Start()
 {
-
+  MutexLockingStrategy::Lock(this->d);
```

Comment:
We need to review the locking strategy here. Framework locks must not be held when calling client code, e.g. the module activator and module changes listeners below.

You can "opt out" and defer this by adding some info about this to the existing threading issues.